gst/nnstreamer/nnstreamer_plugin_api_impl.c

```diff
@@ -956,6 +1019,9 @@ gst_tensor_pad_caps_from_config (GstPad * pad, const GstTensorsConfig * config)
   /* other/tensor */
   if (config->info.num_tensors == 1 && _peer_has_tensor_caps (pad)) {
     caps = _get_tensor_caps (config);
+    if ((peer_caps = gst_pad_peer_query_caps (pad, NULL))) {
+      update_caps_dimension (config, caps, peer_caps);
```

Comment:
line 1023 and 1034
add unref after usage: `gst_caps_unref (peer_caps);`

Reply:
updated! Thank you :)